### ISSUE TITLE
Container Keep Restarting

### ISSUE THREAD

**Issue by albadrun** [ventz/docker-dns-over-https#3]
Hello, I've built this for raspberry pi and the container keep restarting. It keeps showing error like this:
```
Received query from 172.17.0.1 to 0.0.0.0
;; HEADER SECTION
;;	id = 17698
;;	qr = 0	aa = 0	tc = 0	rd = 1	opcode = QUERY
;;	ra = 0	z  = 0	ad = 0	cd = 0	rcode  = NOERROR
;;	qdcount = 1	ancount = 0	nscount = 0	arcount = 0
;;	do = 0

;; QUESTION SECTION (1 record)
;; abs-0.twimg.com.lan.	IN	A

;; ANSWER SECTION (0 records)

;; AUTHORITY SECTION (0 records)

;; ADDITIONAL SECTION (0 records)

Can't use an undefined value as an ARRAY reference at /entrypoint.pl line 60.
```
I think the answer result is `undefined` so the container crashed.

**Comment by ventz** [comment]
@albadrun This is an interesting "cosmetic" bug :)

The DNS portion of the code is still doing the right thing -- since in this case you are feeding it a non-published TLD (.lan), however Perl -- or my lack of error checking -- is erroring out. This is why you are getting a proper response, and then an error. Docker is exiting on the unhandled variable, and thus the container is crashing.

Thanks for picking it up and especially for reporting it.
Fixed and pushed out (1d31e33), and gave you credit for it!

**Comment by albadrun** [comment]
Great, thanks a lot :smile:
